src/v/cloud_storage/partition_manifest.cc

```diff
@@ -348,7 +348,7 @@ partition_manifest::segment_containing(kafka::offset o) const {
         // check using delta_offset_end. If the field is not set then we
         // will return the last segment. This is OK since delta_offset_end
         // will always be set for new segments.
-        if (back->second.committed_kafka_offset() < o) {
+        if (back->second.next_kafka_offset() <= o) {
```

Comment:
This function is also used to get the list of aborted transactions (see `remote_partition::aborted_transactions`).
Let's assume that we call `segment_containing` with offset `o` and the last segment in the manifest has inclusive kafka offsets: `[p, o]`.  If this is a segment written by a previous version of Redpanda, `next_kafka_offset` would yield `o + 1`.  In that case would we miss the aborted transactions for the segment? Granted, that this can only be the case for the last segment in the manifest.

Reply:
In that case, wouldn't we return `back`? Maybe I'm misunderstanding the scenario.

Reply:
I read the diff wrong. The code is correct